ic-helpers/src/factory/state.rs

```diff
@@ -206,21 +206,25 @@ async fn consume_provided_icp(
     Ok(())
 }
 
+#[allow(unused_variables)]
 async fn consume_icp(
     from: Principal,
     amount: u64,
     icp_to: Principal,
     ledger: Principal,
 ) -> Result<(), FactoryError> {
-    LedgerPrincipalExt::transfer(
-        &ledger,
-        icp_to,
-        amount,
-        Some(Subaccount::from(&PrincipalId::from(from))),
-        None,
-    )
-    .await
-    .map_err(FactoryError::LedgerError)?;
+    #[cfg(target_family = "wasm")]
+    {
+        LedgerPrincipalExt::transfer(
+            &ledger,
+            icp_to,
+            amount,
+            Some(Subaccount::from(&PrincipalId::from(from))),
+            None,
+        )
+        .await
+        .map_err(FactoryError::LedgerError)?;
+    }
 
```

Comment:
Same as above comment

Reply:
Again, it'll be more clear when the ledger will be rewritten to have a more elaborate api